### PR TITLE
Move error page to ui

### PR DESCRIPTION
**Pull request by Isaiah-Hamilton**
## What kind of change does this PR introduce?

Moves The error page from `www` and `studio` to the shared ui package

## What is the current behavior?

Two separate error pages that have to be update twice

## What is the new behavior?

One error page that only has to be updated once

### CODE REVIEW

**Comment by MildTomato** [comment]
I'm not sure this is meant to be in the UI package.. 

maybe we should have another package for shared pages/layouts.

maybe we have another directory in ui/src called `/layouts`.

**Comment by Isaiah-Hamilton** [comment]
@MildTomato updated the path to `layouts` and resolved merge conflicts

File: packages/ui/src/layouts/ErrorPage/ErrorPage.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import * as supabaseLogoWordmarkLight from 'common/assets/images/supabase-logo-wordmark--light.png'
+
+export const ErrorPage = (isDarkMode: any) => {
+  const [show404, setShow404] = useState<boolean>(false)
+
+  useEffect(() => {
+    setTimeout(() => {
+      setShow404(true)
+    }, 500)
+  }, [])
```
Comment (saltcod) [inline packages/ui/src/layouts/ErrorPage/ErrorPage.tsx:16]: run a cleanup on this guy?
return () => clearTimeout(timeoutId)

File: apps/www/pages/404.tsx [hunk 2]
```diff
@@ -1,77 +1,12 @@
...
-                  Head back
-                </Button>
-              </a>
-            </Link>
-          </div>
-        </div>
-      </div>
-    </DefaultLayout>
-  )
+  return <ErrorPage isDarkMode={isDarkMode} />
```
Comment (saltcod) [inline apps/www/pages/404.tsx:9]: shouldn't need to pass isDarkMode down? That's in packages/common/, should be able to use it from there?

File: packages/ui/src/layouts/ErrorPage/ErrorPage.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import Image from 'next/image'
+import { useEffect, useState } from 'react'
+import Link from 'next/link'
+import { Button } from 'ui'
+
```
Comment (saltcod) [inline packages/ui/src/layouts/ErrorPage/ErrorPage.tsx:5]: import { useTheme } from 'common'

File: packages/ui/src/layouts/ErrorPage/ErrorPage.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import * as supabaseLogoWordmarkDark from 'common/assets/images/supabase-logo-wordmark--dark.png'
+import * as supabaseLogoWordmarkLight from 'common/assets/images/supabase-logo-wordmark--light.png'
+
+export const ErrorPage = (isDarkMode: any) => {
+  const [show404, setShow404] = useState<boolean>(false)
```
Comment (saltcod) [inline packages/ui/src/layouts/ErrorPage/ErrorPage.tsx:10]: const { isDarkMode } = useTheme()

**Comment by saltcod** [comment]
Hey @Isaiah-Hamilton left a couple of comments & looks like a fresh merge conflict, when you get a chance

File: studio/pages/404.tsx [hunk 5]
```diff
@@ -11,62 +8,7 @@ const Error404: NextPage = ({}) => {
   const { ui } = useStore()
   const { theme } = ui
 
-  const [show404, setShow404] = useState<boolean>(false)
```
Comment (MildTomato) [inline studio/pages/404.tsx:14]: @Isaiah-Hamilton can we revert these changes and let the dashboard team deal with it.
looks like you are removing an important useEffect here.
